components/Promotion/basic_usage.rst

```diff
@@ -0,0 +1,148 @@
+Basic Usage
+===========
+
+.. _component_promotion_generator_coupon-generator:
+
+CouponGenerator
+---------------
+
+In order to automate the process of coupon generation the component provides us with a Coupon Generator.
+
+.. code-block:: php
+
+    <?php
+
+    use Sylius\Component\Promotion\Model\Promotion;
+    use Sylius\Component\Promotion\Generator\Instruction;
+    use Sylius\Component\Promotion\Generator\CouponGenerator;
+
+    $promotion = new Promotion();
+
+    $instruction = new Instruction(); // $amount = 5 by default
+
+    /**
+     * @param RepositoryInterface    $repository
+     * @param EntityManagerInterface $manager
+     */
+    $generator = new CouponGenerator($repository, $manager);
+
+    //This will generate and persist 5 coupons into the database
+    //basing on the instruction provided for the given promotion object
+    $generator->generate($promotion, $instruction);
+
+    // We can also generate one unique code, and assign it to a new Coupon.
+    $code = $gen->generateUniqueCode();
+    $coupon = new Coupon();
+    $coupon->setCode($code);
+
+
+.. _component_promotion_checker_promotion-eligibility-checker:
+
+PromotionEligibilityChecker
+---------------------------
+
+.. code-block:: php
+
+    <?php
+
+    use Sylius\Component\Promotion\Model\Promotion;
+    use Sylius\Component\Core\Model\OrderItem;
+    use Sylius\Component\Promotion\Model\Rule;
+    use Sylius\Component\Promotion\Model\Action;
+    use Sylius\Component\Promotion\Checker\PromotionEligibilityChecker;
+
+    /**
+     * @param ServiceRegistryInterface $registry
+     * @param EventDispatcherInterface $dispatcher
+     */
+    $checker = new PromotionEligibilityChecker($registry, $dispatcher);
```

Comment:
Maybe would be good to show how to register rules in the registry. Otherwise this is a bit confusing, how does the system know about the rules? Or maybe not show, just add a note here that you need the actions registered.